napari/layers/labels/_labels_mouse_bindings.py

```diff
@@ -0,0 +1,37 @@
+from ._labels_utils import interpolate_coordinates
+
+
+def paint(layer, event):
+    """Paint with the currently selected label."""
+    # on press
+    layer._save_history()
+    layer._block_saving = True
+    layer.paint(layer.coordinates, layer.selected_label)
+    last_cursor_coord = layer.coordinates
+    yield
+
+    # on move
+    while event.type == 'mouse_move':
```

Comment:
I found this **highly** confusing.  It looks like it's a vispy implementation detail that it modifies the original mouse press event to `mouse_move` and later to `mouse_release`.  I couldn't see anything in the documentation that explains this behavior.  It makes me squeamish. :/

I would much much rather observe the current event than to depend on this quirk.  You could plausibly [send data back to the generator and save it](https://docs.python.org/3/reference/expressions.html#generator.send).

Reply:
Agreed, @kne42 and I were actually talking about this yesterday - she's planning a refactor to the mouse functions that will move us away from vispys event system into our own. Maybe she can talk about it at the dev meeting today, or can get your feedback on how she is thinking about it. As this will come later, I'd say we still merge as is and then fix during that refactor

Reply:
SGTM.

I would caution against using generators for this though.  It's not the most intuitive if you want developers to extend mouse clicking functionality, and error handling appears to be quite brittle.